lib/rules/html-self-closing-style.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * @author Toru Nagashima
+ * @copyright 2016 Toru Nagashima. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+/**
+ * Kind strings.
+ * This strings wil be displayed in error messages.
+ */
+const KIND = Object.freeze({
+  NORMAL: 'HTML elements',
+  VOID: 'HTML void elements',
+  COMPONENT: 'Vue.js custom components',
+  SVG: 'SVG elements',
+  MATH: 'MathML elements'
+})
+
+/**
+ * Normalize the given options.
+ * @param {Object|undefined} options The raw options object.
+ * @returns {Object} Normalized options.
+ */
+function parseOptions (options) {
+  return {
+    [KIND.NORMAL]: (options && options.html && options.html.normal) || 'never',
+    [KIND.VOID]: (options && options.html && options.html.void) || 'never',
+    [KIND.COMPONENT]: (options && options.html && options.html.component) || 'always',
+    [KIND.SVG]: (options && options.svg) || 'always',
+    [KIND.MATH]: (options && options.math) || 'always'
+  }
+}
+
+/**
+ * Get the kind of the given element.
+ * @param {VElement} node The element node to get.
+ * @returns {string} The kind of the element.
+ */
+function getKind (node) {
+  if (utils.isCustomComponent(node)) {
+    return KIND.COMPONENT
+  }
+  if (utils.isHtmlElementNode(node)) {
+    if (utils.isHtmlVoidElementName(node.name)) {
+      return KIND.VOID
+    }
+    return KIND.NORMAL
+  }
+  if (utils.isSvgElementNode(node)) {
+    return KIND.SVG
+  }
+  if (utils.isMathMLElementNode(node)) {
+    return KIND.MATH
+  }
+  return 'unknown elements'
+}
+
+/**
+ * Check whether the given element is empty or not.
+ * This ignores whitespaces.
+ * @param {VElement} node The element node to check.
+ * @returns {boolean} `true` if the element is empty.
+ */
+function isEmpty (node) {
+  return node.children.every(child => child.type === 'VText' && child.value.trim() === '')
```

Comment:
You are checking only if element has VText and its not empty and you should also check if there are other elements inside. with somethink like this:
`child.type !== 'VText' || child.value.trim() === ''`

e.g. `html.component = allways`
`<component />` - it will not report error now
`<component><div /></component>` - it will report error now